pkg/kubectl/resize.go

```diff
@@ -159,9 +158,33 @@ func (resizer *ReplicationControllerResizer) Resize(namespace, name string, newS
 	if waitForReplicas != nil {
 		rc := &api.ReplicationController{ObjectMeta: api.ObjectMeta{Namespace: namespace, Name: name}}
 		if err := wait.Poll(waitForReplicas.interval, waitForReplicas.timeout,
-			client.ControllerHasDesiredReplicas(resizer, rc)); err != nil {
+			resizer.c.ControllerHasDesiredReplicas(rc)); err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+// ResizerClient abstracts access to ReplicationControllers.
+type ResizerClient interface {
+	GetReplicationController(namespace, name string) (*api.ReplicationController, error)
+	UpdateReplicationController(namespace string, rc *api.ReplicationController) (*api.ReplicationController, error)
+	ControllerHasDesiredReplicas(rc *api.ReplicationController) wait.ConditionFunc
+}
+
+// RealResizerClient is a ResizerClient which uses a Kube client.
+type RealResizerClient struct {
```

Comment:
I sort of hate that this is a public symbol.  Could we maybe define it in an internal pkg (using Go's internal trickery) to make shared across kubectl but not the whole universe?